BrainPortal/lib/boutiques_create_fake_parent_dir.rb

```diff
@@ -0,0 +1,152 @@
+  # filename CBRAIN will create a valid one instead.
+  def descriptor_for_form #:nodoc:
+    descriptor                = super.dup
+    parent_dirname_by_inputid = descriptor.custom_module_info('BoutiquesCreateFakeParentDir')
+
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      fake_parent_dirname = fake_parent_dirname[0] if fake_parent_dirname.is_a?(Array)
+      if !Userfile.is_legal_filename?(fake_parent_dirname)
+        fake_parent_dirname = "fake_dir_uniq_id"
+      end
+
+      # Adjust the description
+      input              = descriptor.input_by_id(inputid)
+      input.description  = input.description.to_s +
+                           "\nThis input will be copied in a parent folder: #{fake_parent_dirname}.The parent folder will be used in the command line"
+    end
+
+    descriptor
+  end
+
+  ############################################
+  # Bourreau (Cluster) Side Modifications
+  ############################################
+
+  # For input in `BoutiquesCreateFakeParentDir` section,
+  # create a fake parent directory that will contains a symlink
+  # to the orginal selected Userfile.
+  def setup #:nodoc:
+
+    descriptor = self.descriptor_for_setup
+    self.addlog(descriptor.file_revision_info.format("%f rev. %s %a %d"))
+
+    parent_dirname_by_inputid = descriptor.custom_module_info('BoutiquesCreateFakeParentDir')
+
+    # Remove IDs from invoke_params
+    params["BoutiquesCreateFakeParentDir"] = {}
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      params["BoutiquesCreateFakeParentDir"][inputid] = invoke_params[inputid]
+      invoke_params[inputid] = nil
+    end
+
+    # invoke main setup
+    result = super
+    return false if !super
+
+    # Re-insert IDs in invoke_params
+    invoke_params.each do |inputid, userfile_id|
+      next if userfile_id.present?
+      invoke_params[inputid] = params["BoutiquesCreateFakeParentDir"][inputid]
+    end
+
+    # Special make_available who need to have a parent folder
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      userfile_id = params["BoutiquesCreateFakeParentDir"][inputid]
+
+      params['BoutiquesCreateFakeParentDir'][inputid] = nil if userfile_id.blank?
+      next if userfile_id.blank?
+
+      userfile    = Userfile.find(userfile_id)
+
+      if !Userfile.is_legal_filename?(fake_parent_dirname)
+        fake_parent_dirname = "fake_dir_#{userfile_id}"
+      end
+
+      make_available(userfile, "#{fake_parent_dirname}/#{userfile.name}")
+      parent_dirname_by_inputid[inputid]              = fake_parent_dirname
+      params['BoutiquesCreateFakeParentDir'][inputid] = fake_parent_dirname
+    end
+
+    true
+  end
+
+  # Overrides the same method in BoutiquesClusterTask, as used
+  # during cluster_commands()
+  def finalize_bosh_invoke_struct(invoke_struct) #:nodoc:
+    self.params['BoutiquesCreateFakeParentDir'].each do |inputid, dirname|
```

Comment:
super() is a method call, and every time you'll be editing potentially a different hash; you cannot have a loop that has super() in it, you're supposed to just call it once, override it's content and return the new content.